src/lib/helpers/renderAs.tsx

```diff
@@ -0,0 +1,119 @@
+import classNames from 'classnames';
+import { cloneElement, ComponentProps, createElement, ElementType, forwardRef, Fragment, isValidElement } from 'react';
+import { mergeRefs } from 'react-merge-refs';
+
+export type Props<TTag extends ElementType> = Omit<ComponentProps<TTag>, 'ref'> & { as?: TTag };
+
+export interface RederAsProps<TTag extends ElementType> {
+  defaultTag: ElementType;
+  componentName: string;
+  passedProps: Props<TTag>;
+}
+
+interface PropChildren {
+  children?: React.ReactNode;
+  className?: string;
+  style?: React.CSSProperties;
+  ref?: React.Ref<unknown>;
+}
+
+export function RenderAs<TTag extends ElementType>({
```

Comment:
I like the idea of this component. My two cents here:

- Rename the filename to `RenderAs.tsx`
- Create some unit tests to test it